### PR TITLE
[Feat] Add patching speed check

### PR DESCRIPTION
**Pull request by BrettCleary**
# Summary

Throw if patching will take longer than the download.

Using rough estimate numbers to get some estimates.

Log with "Patching Too Slow" instead of "Patching Error" so metrics graphs don't see it as a failed event.
- Introduces a new patching error object to accomplish this

closes https://github.com/HyperPlay-Gaming/product-management/issues/755

### CODE REVIEW

File: src/backend/storeManagers/hyperplay/games.ts [hunk 1]
```diff
@@ -1653,6 +1654,82 @@ export async function downloadGameIpdtManifest(
...
+    )
+    const downloadSize = parseInt(
+      releaseMeta.platforms[platform]?.downloadSize ?? '0'
+    )
+    const installSize = parseInt(
+      releaseMeta.platforms[platform]?.installSize ?? '0'
+    )
+    // @TODO: get these speed values from local checks of download/write speed
+    const downloadSpeedInKBPerSecond = 25 * 1024
+    const extractionSpeedInKBPerSecond = 50 * 1024
```
Comment (BrettCleary) [inline src/backend/storeManagers/hyperplay/games.ts:1695]: open to suggestions here. I'd suggest we leave for another future iteration though

**Review by jiyuu-jin — APPROVED** [review]
This looks great, testing now!

File: src/backend/storeManagers/hyperplay/games.ts [hunk 2]
```diff
@@ -1653,6 +1654,82 @@ export async function downloadGameIpdtManifest(
...
+    const estTimeToInstallFullGameInSec =
+      (downloadSize / 1024) * downloadSpeedInKBPerSecond +
+      (installSize / 1024) * extractionSpeedInKBPerSecond
+
+    // @TODO: get this value from local check of patching speed
+    const patchingSpeedEstimateInKBPerSecond = 5 * 1024
```
[the remark this answers is not in the document]
Comment (flavioislima) [reply src/backend/storeManagers/hyperplay/games.ts:1701]: Maybe add an object with these amounts and we can import them from LD.
This way we can alter any of them to make the calculation more precise.

**Review by flavioislima — APPROVED** [review]
LGTM

**Review by eliobricenov — APPROVED** [review]

**Review by nyghtstalker — APPROVED** [review]
In order for the new patching to start I had to delete the old manifest version 0 so the client could download version 1. After, I was able to confirm the updated patching and noticed the improved speed.

image

**Pull request merged**